spec/complex_type_spec.rb

```diff
@@ -310,4 +310,16 @@ def make_bar
     type = return_type.resolve_generics(generic_class, called_method.return_type)
     expect(type.tag).to eq('Array<String>')
   end
+
+  it 'parses tuples of tuples' do
+    api_map = Solargraph::ApiMap.new
+    type = Solargraph::ComplexType.parse('Array(Array(String), String)')
+    expect(type.tag).to eq('Array(Array(String), String)')
+  end
+
+  it 'parses tuples of tuples with same type twice in a row' do
+    api_map = Solargraph::ApiMap.new
+    type = Solargraph::ComplexType.parse('Array(Symbol, String, Array(Integer, Integer))')
+    expect(type.to_s).to eq('Array(Symbol, String, Array(Integer, Integer))')
```

Comment:
The (Integer, Integer) here does collapse into (Integer) at some point between here and appearing in my LSP results.  I'll chase this down next and either add this to the same PR or push a new one if you get to this first.

Reply:
Ah, the qualify() method in UniqueType was applying .uniq, assuming that every list of types represented  a union type.  Fixed.